src/main/java/uk/gov/companieshouse/pscstatementdataapi/api/CompanyMetricsApiService.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.companieshouse.pscstatementdataapi.api;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.handler.metrics.request.PrivateCompanyMetricsGet;
+import uk.gov.companieshouse.api.metrics.MetricsApi;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.logging.Logger;
+
+import java.util.Optional;
+
+
+@Service
+public class CompanyMetricsApiService {
+
+    private static final String GET_COMPANY_METRICS_ENDPOINT = "/company/%s/metrics";
+    private final Logger logger;
+
+
+    private ApiClientService apiClientService;
+
+    /**
+     * Invoke Company Metrics API.
+     */
+    @Autowired
+    public CompanyMetricsApiService(Logger logger,
+                                    @Qualifier("CompanyMetricsApiClient") ApiClientService apiClientService) {
+        this.logger = logger;
+        this.apiClientService = apiClientService;
+    }
+
+    /**
+     * Get company metrics.
+     *
+     * @param companyNumber company number.
+     * @return company metrics.
+     */
+    public Optional<MetricsApi> getCompanyMetrics(final String companyNumber) {
+        logger.info(String.format("Started : getCompanyMetrics for Company Number %s ",
+                companyNumber
+        ));
+        final InternalApiClient internalApiClient = this.apiClientService.getInternalApiClient();
+        PrivateCompanyMetricsGet companyMetrics =
+                internalApiClient.privateCompanyMetricsResourceHandler()
+                        .getCompanyMetrics(
+                                String.format(GET_COMPANY_METRICS_ENDPOINT, companyNumber));
+        try {
+            ApiResponse<MetricsApi> execute = companyMetrics.execute();
+            return Optional.ofNullable(execute.getData());
+        } catch (URIValidationException exp) {
```

Comment:
Please print exception in log for better trace

Reply:
done